### PR TITLE
Support for per-vlan traps

### PR DESCRIPTION
**Pull request by bandaru-viswanath**
Add an ability for the user to specify a list of traps that are applicable (only) for that vlan. This list is in-addition to the traps that are otherwise generally defined.

### CODE REVIEW

**Comment by rlhui** [comment]
What if same object is in both the host trap table and this vlan trap table?

**Comment by rck-innovium** [comment]
As discussed during the meeting can you summarize the reasons for diverging from the behavior of SAI_HOSTIF_TRAP_ATTR_EXCLUDE_PORT_LIST, i.e. why not have have SAI_HOSTIF_TRAP_ATTR_EXCLUDE_VLAN_LIST?

**Comment by rck-innovium** [comment]
The key to create the SAI_HOSTIF_TRAP object is currently the SAI_HOSTIF_TRAP_ATTR_TRAP_TYPE. The use case where two VLANs that need different behavior for the same trap will not work as is today.

**Comment by rlhui** [comment]
There's a feedback from Matty in sai meeting - packets going to cpu need to be via hostif_trap channel, it cannot be omitted, even though it is meant to be per-vlan trap.  @mattyk , please clarify if any.  @bandaru-viswanath should address this first.

**Comment by JaiOCP** [comment]
> As discussed during the meeting can you summarize the reasons for diverging from the behavior of SAI_HOSTIF_TRAP_ATTR_EXCLUDE_PORT_LIST, i.e. why not have have SAI_HOSTIF_TRAP_ATTR_EXCLUDE_VLAN_LIST?

Would it make sense to have then a common exclude LIST based on a type where TYPE can be PORT/VLAN/RIF ?

**Comment by lguohan** [comment]
retest this please

**Comment by JaiOCP** [comment]
+Vissu

 

This PR will to be closed by the Author.

 

-Jai

**Comment by bandaru-viswanath** [comment]
Closing this PR as per recommendation.
